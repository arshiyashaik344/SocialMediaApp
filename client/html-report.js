var reporter = require('cucumber-html-reporter');
var options = {
  theme: 'bootstrap',
  jsonFile: 'reports/report.json',
  output: 'reports/cucumber_report.html',
  reportSuiteAsScenarios: true,
  scenarioTimestamp: true,
  launchReport: true,
  metadata: {
    Title: 'Sample Report',
    Browser: 'chrome',
    Platform: 'Linux'
  }
};
reporter.generate(options);
