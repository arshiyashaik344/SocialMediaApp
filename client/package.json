{
  "name": "client",
  "version": "0.1.0",
  "private": true,
  "dependencies": {
    "@apollo/client": "^3.4.17",
    "@apollo/react-hooks": "^4.0.0",
    "apollo-link-context": "^1.0.18",
    "apollo-link-ws": "^1.0.20",
    "cucumber-html-reporter": "^5.5.0",
    "gherkin-testcafe": "^5.2.0",
    "graphql": "^15.7.2",
    "graphql-tag": "^2.10.1",
    "jwt-decode": "^2.2.0",
    "moment": "^2.29.1",
    "react": "^17.0.2",
    "react-dom": "^17.0.2",
    "react-paginate": "^8.1.0",
    "react-router-dom": "^5.3.0",
    "react-scripts": "^3.0.1",
    "semantic-ui-css": "^2.4.1",
    "semantic-ui-react": "^2.0.4"
  },
  "scripts": {
    "start": "react-scripts start",
    "dev": "webpack serve",
    "build": "webpack",
    "test": "./node_modules/@cucumber/cucumber/bin/cucumber-js --format json:./reports/report.json",
    "eject": "react-scripts eject",
    "lint": "eslint .",
    "lint:fix": "eslint --fix",
    "format": "prettier --write './**/*.{js,jsx,ts,tsx,css,md,json}' --config ./.prettierrc"
  },
  "eslintConfig": {
    "extends": "react-app"
  },
  "browserslist": {
    "production": [
      ">0.2%",
      "not dead",
      "not op_mini all"
    ],
    "development": [
      "last 1 chrome version",
      "last 1 firefox version",
      "last 1 safari version"
    ]
  },
  "devDependencies": {
    "@babel/core": "^7.16.7",
    "@babel/plugin-syntax-jsx": "^7.16.7",
    "@babel/preset-env": "^7.16.7",
    "@babel/preset-react": "^7.16.7",
    "@cucumber/cucumber": "^7.3.2",
    "@testing-library/jest-dom": "^5.16.1",
    "@testing-library/react": "^12.1.2",
    "@testing-library/react-hooks": "^7.0.2",
    "@testing-library/user-event": "^13.5.0",
    "babel-jest": "^27.4.6",
    "babel-loader": "^8.2.3",
    "css-loader": "^6.5.1",
    "eslint": "^7.32.0",
    "eslint-config-prettier": "^8.3.0",
    "eslint-plugin-prettier": "^4.0.0",
    "eslint-plugin-react": "^7.28.0",
    "html-webpack-plugin": "^4.5.2",
    "jest": "^27.5.1",
    "jest-environment-jsdom": "^27.4.6",
    "prettier": "^2.5.1",
    "react-test-renderer": "^17.0.2",
    "resolve-url-loader": "^5.0.0",
    "sass-loader": "^12.4.0",
    "style-loader": "^3.3.1",
    "url-loader": "^4.1.1",
    "vue-jest": "^3.0.7",
    "webpack": "^5.65.0",
    "webpack-cli": "^4.9.1",
    "webpack-dev-server": "^4.7.4"
  },
  "jest": {
    "verbose": true,
    "testEnvironment": "jsdom"
  }
}
