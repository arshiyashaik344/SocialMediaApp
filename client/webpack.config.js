const path = require('path');
const HtmlWebpackPlugin = require('html-webpack-plugin');

module.exports = {
  entry: path.join(__dirname, 'src', 'index.js'),
  output: {
    path: path.resolve(__dirname, 'dist')
  },
  mode: 'development',
  module: {
    rules: [
      {
        test: /\.?js$/,
        exclude: /node_modules/,
        use: {
          loader: 'babel-loader',
          options: {
            presets: [
              '@babel/preset-env',
              '@babel/preset-react',
              {
                plugins: ['@babel/plugin-syntax-jsx']
              }
            ]
          }
        }
      },
      {
        test: /\.css$/i,
        use: [
          {
            loader: 'style-loader'
          },
          {
            loader: 'css-loader',
            options: {
              modules: true,
              sourceMap: true
            }
          },
          {
            loader: 'resolve-url-loader',
            options: {
              removeCR: true
            }
          }
        ],
        include: [path.join(__dirname, 'src'), /node_modules/]
      },
      // {
      //   test: /\.s[a|c]ss$/,
      //   use: [
      //     {
      //       loader: 'sass-loader'
      //     },
      //     {
      //       loader: 'style-loader'
      //     },
      //     {
      //       loader: 'css-loader'
      //     }
      //   ],
      //   include: [path.join(__dirname, 'src'), /node_modules/]
      // },
      {
        test: /\.(png|jpg|woff|svg|eot|ttf|woff2)$/,
        use: {
          loader: 'url-loader',
          options: {
            limit: 1000,
            name: 'assets/img/[name].[ext]'
          }
        }
      }
    ]
  },
  plugins: [
    new HtmlWebpackPlugin({
      template: path.join(__dirname, 'public', 'index.html')
    })
  ]
};
