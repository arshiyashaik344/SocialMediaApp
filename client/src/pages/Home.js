import React, { useContext, useState } from 'react';
import { useQuery } from '@apollo/client';
import { Grid, Transition } from 'semantic-ui-react';
import ReactPaginate from 'react-paginate';

import { AuthContext } from '../context/auth';
import PostCard from '../components/PostCard';
import PostForm from '../components/PostForm';
import { FETCH_POSTS_QUERY } from '../util/graphql';

function Home() {
  const { user } = useContext(AuthContext);
  const {
    loading,
    data: { getPosts: posts } = {}
  } = useQuery(FETCH_POSTS_QUERY);

  //Logic for pagination (19-27)
  const [pageNumber, setPageNumber] = useState(0);
  const postsPerPage = 9;
  const pagesVisited = pageNumber * postsPerPage;
  
  const pageCount = posts ? Math.ceil(posts.length / postsPerPage) : '';

  const changePage = ({selected}) => {
    setPageNumber(selected);
  }

  return (
    <>
    <Grid columns={3}>
      <Grid.Row className="page-title">
        <h1>Recent Posts</h1>
      </Grid.Row>
      <Grid.Row>
        {user && (
          <Grid.Column>
            <PostForm />
          </Grid.Column>
        )}
        {loading ? (
          <h1>Loading posts..</h1>
        ) : (
          <Transition.Group>
            {posts &&
              posts.slice(pagesVisited, pagesVisited + postsPerPage).map((post) => (
                <Grid.Column key={post.id} style={{ marginBottom: 20 }}>
                  <PostCard post={post} />
                </Grid.Column>
              ))}
              
          </Transition.Group>
        )}
      </Grid.Row>
    </Grid>
    {loading ? ' ' : 
      <ReactPaginate 
      previousLabel = 'Previous'
      nextLabel = "Next"
      pageCount={pageCount}
      onPageChange= {changePage}
      containerClassName='paginationBttns'
      previousLinkClassName='previousBttn'
      nextLinkClassName='nextBttn'
      disabledClassName='paginationDisabled'
      activeClassName='paginationActive'
    />
    }
    </>
  );
}

export default Home;
