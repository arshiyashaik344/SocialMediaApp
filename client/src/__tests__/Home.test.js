/* eslint-disable no-unused-vars */
import { screen } from '@testing-library/react';
import React from 'react';
import TestRenderer from 'react-test-renderer';
import { MockedProvider } from '@apollo/client/testing';

import Home from '../pages/Home';

const mocks = [];

describe('Home Component', () => {
  test('rendered input', () => {
    const component = TestRenderer.create(
      <MockedProvider mocks={mocks} addTypename={false}>
        <Home />
      </MockedProvider>
    );

    const tree = component.toJSON();
    // console.log(tree);
    // console.log(component.root);
    expect(tree).toMatchSnapshot();
    const instance = component.root;
    instance.findByProps({ className: 'ui three column grid' }).exists;
    const heading = instance.findByProps({
      className: 'ui three column grid'
    }).children[0];
    expect(heading.findByProps({ className: 'heading' }).children).toEqual([
      'Recent Posts'
    ]);
  });
});

// **** Generic && tried tests ********
// expect(2 + 2).toBe(4);
// console.log(heading.findByProps({ className: 'heading' }).children);

// console.log(heading);
// console.log(instance);
// console.log(
//   instance.findByProps({ className: 'ui three column grid' }).children
// );

// expect(
//   instance.findByProps({ className: 'ui three column grid' }).children
// ).toEqual('Recent Posts');
// render(<Home />);
// jest.spyOn(window.localStorage.__proto__, 'setItem');
// window.localStorage.__proto__.setItem = jest.fn();
// assertions as usual:
// expect(localStorage.setItem).toHaveBeenCalled();

// const heading = screen.getByText('Recent Posts');
// expect(heading).toBeInTheDocument();
// screen.debug();
