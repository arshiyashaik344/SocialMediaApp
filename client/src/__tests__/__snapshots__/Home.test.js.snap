// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Component rendered input 1`] = `
Array [
  <div
    className="ui three column grid"
    id="posts-grid"
  >
    <div
      className="row page-title"
    >
      <h1
        className="heading"
        id="posts"
      >
        Recent Posts
      </h1>
    </div>
    <div
      className="row"
    >
      <h1
        className="posts"
      >
        Loading posts..
      </h1>
    </div>
  </div>,
  " ",
]
`;
